### PR TITLE
fixes issue #410

### PR DESCRIPTION
**Pull request by harschware**
the build problem described in issue #410 was due to a very out of date setup tools.  At one point in testing I got a warning about pip out of date so I upgrade that too...

### CODE REVIEW

File: Dockerfile.jupyter [hunk 1]
```diff
@@ -9,6 +9,9 @@ RUN conda install requests-kerberos -y
 
 USER $NB_USER
 
+RUN pip install --upgrade pip
+RUN pip install --upgrade --ignore-installed setuptools
```
Comment (aggFTW) [inline Dockerfile.jupyter]: Is the `--ignore-installed` needed?

Comment (harschware) [reply Dockerfile.jupyter]: yes. it was required.

Comment (aggFTW) [reply Dockerfile.jupyter]: Why?

File: Dockerfile.spark [hunk 2]
```diff
@@ -10,6 +10,8 @@ RUN apt-get update && apt-get install -yq --no-install-recommends --force-yes \
     r-base-core && \
     rm -rf /var/lib/apt/lists/*
 
+RUN pip install --upgrade setuptools
```
Comment (aggFTW) [inline Dockerfile.spark]: Why not use ignore installed here too then? Should we update pip here as well?

Comment (harschware) [reply Dockerfile.spark]: --ignore-installed was not needed here.  I don't know why.  My environment is OSX Sierra with Docker Community Edition Version 17.09.0-ce-mac35 (19611)

As for upgrading pip here..  this image is built via `FROM gettyimages/spark:2.1.0-hadoop-2.7` which had pip up to date.  Adding the upgrade will yield:
Step 3/12 : RUN pip install --upgrade pip
 ---> Running in 58351e04718b
Requirement already up-to-date: pip in /usr/local/lib/python3.4/dist-packages/pip-9.0.1-py3.4.egg
 whereas the jupyter notebook is built via `FROM jupyter/base-notebook:d0b2d159cc6c` and that pinned version is not up to date.

With this image built from an unpinned version of gettyimages, we may be able to defer the pip version to the base docker image.

File: Dockerfile.jupyter (hunk 1 above)
Comment (harschware) [reply Dockerfile.jupyter]: without it I got this error:  

```Step 7/25 : RUN pip install --upgrade setuptools
 ---> Running in a103e27aaeca
Collecting setuptools
  Downloading setuptools-36.6.0-py2.py3-none-any.whl (481kB)
Installing collected packages: setuptools
  Found existing installation: setuptools 23.0.0
Cannot remove entries from nonexistent file /opt/conda/lib/python3.5/site-packages/easy-install.pth
The command '/bin/sh -c pip install --upgrade setuptools' returned a non-zero code: 1
```

I googled until I found the --ignore-installed could get around it.  See:   https://github.com/ContinuumIO/anaconda-issues/issues/542

File: Dockerfile.spark (hunk 2 above)
Comment (harschware) [reply Dockerfile.spark]: Hi @aggFTW, can you review again?

**Comment by aggFTW** [comment]
Done and thanks so much for the explanations!

**Pull request merged**